### PR TITLE
Feature/mlme beacon notify

### PR DESCRIPTION
**Pull request by gc100**

### CODE REVIEW

File: src/nl802154.h [hunk 1]
```diff
@@ -112,6 +117,33 @@ enum nl802154_attrs {
 
 	NL802154_ATTR_SUPPORTED_COMMANDS,
 
+	NL802154_ATTR_BEACON_INDICATION_TIMEOUT,
```
Comment (cfriedt) [inline src/nl802154.h:139]: Please put additional attributes at the bottom of the list, where it says "/\* add attributes here, update the policy in nl802154.c */"

File: src/phy.c [hunk 2]
```diff
@@ -338,3 +338,164 @@ static int handle_ed_scan(struct nl802154_state *state,
 
 COMMAND(get, ed_scan, "[<page> [<channels> [<duration>]]]",
     NL802154_CMD_ED_SCAN_REQ, 0, CIB_PHY, handle_ed_scan, NULL);
+
+
+static int print_beacon_notify_indication(struct nl_msg *msg, void *arg)
```
Comment (cfriedt) [inline src/phy.c]: rename to nl802154_beacond_notifiy_ind

File: src/phy.c [hunk 3]
```diff
@@ -338,3 +338,164 @@ static int handle_ed_scan(struct nl802154_state *state,
...
+	printf("beacon_indication:\n");
+
+	if (tb_msg[NL802154_ATTR_BEACON_SEQUENCE_NUMBER]) {
+		printf("BSN: %x\n", nla_get_u32(tb_msg[NL802154_ATTR_BEACON_SEQUENCE_NUMBER]));
+	}
+
+	if (tb_msg[NL802154_ATTR_PAN_DESCRIPTOR]) {
+		struct nlattr *tb_pan_desc[NL802154_ATTR_MAX + 1];
+
+		static struct nla_policy pan_desc_policy[NL802154_ATTR_MAX + 1] = {
```
Comment (cfriedt) [inline src/phy.c:377]: put the policy in nl802154_policy

File: src/phy.c [hunk 4]
```diff
@@ -338,3 +338,164 @@ static int handle_ed_scan(struct nl802154_state *state,
...
+	return NL_SKIP;
+}
+
+static int handle_beacon_notify(struct nl802154_state *state,
+				 struct nl_cb *cb,
+				 struct nl_msg *msg,
+				 int argc, char **argv,
+				 enum id_input id)
+{
+    printf("Inside %s %d\n", __FUNCTION__,  atoi(argv[0]) );
```
Comment (cfriedt) [inline src/phy.c]: remove printf

File: src/phy.c [hunk 5]
```diff
@@ -338,3 +338,164 @@ static int handle_ed_scan(struct nl802154_state *state,
...
+	NLA_PUT_U32(msg, NL802154_ATTR_BEACON_INDICATION_TIMEOUT, atoi(argv[0]));
+	nl_cb_set(cb, NL_CB_VALID, NL_CB_CUSTOM, print_beacon_notify_indication, NULL);
```
Comment (cfriedt) [inline src/phy.c]: pretty sure the return value from nl_cb_set is being ignored.

You also need to add scheduled work to timeout and cancel the work. 

Please use existing code from any of the other branches as an example

File: src/phy.c (hunk 3 above)
Comment (gc100) [inline src/phy.c:377]: Are you referring to the nl802154_policy in the kernel?  I was following the method used in print_phy_handler() in src/info.c

File: src/phy.c (hunk 5 above)
Comment (gc100) [inline src/phy.c]: Will push a change to handle return code from nl_cb_set.

I will send a pull request for kernel code timeout change.

File: src/phy.c [hunk 6]
```diff
@@ -338,3 +338,176 @@ static int handle_ed_scan(struct nl802154_state *state,
 
 COMMAND(get, ed_scan, "[<page> [<channels> [<duration>]]]",
     NL802154_CMD_ED_SCAN_REQ, 0, CIB_PHY, handle_ed_scan, NULL);
+
+
+static int print_nl802154_beacon_notify_ind(struct nl_msg *msg, void *arg)
```
Comment (cfriedt) [inline src/phy.c:343]: I would just rename this to "handle_beacon_notify" to line up with the rest of the code.

File: src/phy.c (hunk 3 above)
Comment (cfriedt) [inline src/phy.c:377]: Oh yeah, sorry. You don't even need policy in here - just remove it entirely and replace references to it to NULL (all of the other code does the same).

File: src/phy.c (hunk 6 above)
Comment (gc100) [inline src/phy.c:343]: handle_beacon_notify() is already the iwpan command handler.  I have been using the print_xxx for the callbacks that print received data to the terminal.  eg. print_ed_scan_handler().

Comment (cfriedt) [inline src/phy.c:343]: As long as it matches the naming convention that the rest of the callbacks use (e.g. see phy.c).

**Comment by cfriedt** [comment]
There are a few outstanding issues, so I'm closing this request for now.

**Pull request closed without merging**